### ISSUE TITLE
Document Label Marking

### ISSUE THREAD

**Issue by brianrufgsa** [usnistgov/OSCAL#600]
# User Story:

As an OSCAL content creator, I need a way to indicate the document label that would appear if OSCAL content is rendered as a document, as often required to indicate information sensitivity or ownership. Similar to what is traditionally done in documents on title pages and page header/footer areas. 

Examples of commonly required document markings include:

- Controlled Unclassified Information (CUI)
- Company Proprietary
- Company Sensitive and Proprietary

DoD or the IC may require:
- RESTRICTED
- CONFIDENTIAL
- SECRET
- TOP SECRET

## Goals:

Establish syntax within metadata that enables content creators to attach document sensitivity markings to OSCAL content.

## Dependencies:

Related to Issue #597 

## Acceptance Criteria

- [ ] All [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes in this issue have been updated. Changes to the OSCAL website can be made in the docs/content directory of your branch.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

- [ ]  A standard mechanism exists that enables a content creator to add one or more required sensitivity designations to an OSCAL file.

**Comment by brianrufgsa** [comment]
In the _Guide to OSCAL-based FedRAMP System Security Plans_, I defined the following extension to address this issue:

```
<metadata>
   <!-- cut: title, published, last-modified, version, oscal-version -->
   <prop name="marking" ns="https://fedramp.gov/ns/oscal">Controlled Unclassified Information</prop>
</metadata>
```

**Comment by bradh** [comment]
If we need element level marking, then it probably needs to be an attribute (which `prop` can't do). If its more section level, that is OK.

For the IC (including DoD), its almost always IC-ISM. Its a lot more complicated than what we have here though. NATO has STANAG 4774, which is an implementation (perhaps) of XML SPIF http://www.xmlspif.org/ which is a way to describe the policy.

I worry that this is a deep rabbit hole.

**Comment by oscalbuilder** [comment]
@bradh Seems like a SPIF can be a source of marking definitions.

Can you elaborate on what you mean by "then it probably needs to be an attribute (which prop can't do)"?

I am also thinking about this as to how it relates to JSON, which will need to be JSON object level marking. Most of our assemblies allow for prop. There are some supporting assemblies (e.g., description, remarks) which don't.

**Comment by redhatrises** [comment]
Having the classification in the document metadata makes a lot of sense as long as it is not duplicated in the metadata and it remains free-flowing text i.e. don't specify what the only type of data classification is allowed. Using an attribute creates a bigger problem in that it removes the classification from the actual text in the descriptions, remarks, etc.

**Comment by bradh** [comment]
If you are working on a system (or in an environment) where portion marking is required, then you need to mark at the paragraph level.

The document title might be "System Name (U)" or "System Name (FOUO)".  

A table might have an unclas caption, but the content might be classified higher. Or a single row or single cell in the table might be Confidential. Or it might be TS and compartmented, and rel to a different set of countries. Or it might be FGI.  And you need declas dates. IC-ISM deals with that by using attributes.

However not all countries or all consumers care about security markings. Some are happy at the document level. Those that do care about portion marking / tearlines don't all use IC-ISM (actually, only one does, but its a big user...). Also the current version of the IC-ISM schema is not public. At the other extreme, the 4774 approach is so open you can plugin whatever you want, but you end up needing another profile for what is valid.

**Comment by brianrufgsa** [comment]
Taking the crawl, walk, run approach I'd like to suggest that at least getting a document marking field in metadata is "crawl" and likely applies to over 90% of potential OSCAL users. 
Paragraph level marking is more like "walk", and probably applies to 20-30% of potential users. 

I agree we eventually need both. There is a focused effort on metadata and back-matter changes, which is why I raised this now. At least we can get the document-level marking in place as part of that Epic.  It's low effort and high value.

Ultimately, I think we need to better understand how paragraph-level marking typically applies to security authorization documents (SSPs, SAP, SAR) and how it applies to non-document-based data (such as in a database). While I have encountered such marking in other settings, my experience is limited. 

As I understand it, even if the entire document is classified at the same level, paragraph-level markings may still be necessary (and are all going to be marked the same), but I _believe_ those rules apply to documents, not data. 

In other words, I _believe_ an OSCAL file is more analogous to a database, and it isn't until that OSCAL data is presented in a printable format (ie rendered as PDF, MS Word or HTML) that those paragraph markings become required. I'd love to hear a more qualified interpretation on this.

@bradh, can you live with just getting the document-level marking captured now in metadata, and then we can create a separate issue to address paragraph-level marking?

**Comment by bradh** [comment]
Conversely, whole document marking (at the `metadata` level) would probably still be required even if there was paragraph markings. So OK with this as a start.

I would caution that its probably worth adding some kind of qualifier to the sensitivity marking, so I can say what kind of sensitivity marking I mean. In particular "confidential" is both a security marking in the DoD / IC sense, and a common intellectual property marking for commercial organisations, and while they are both sensitivity related, the handling for those looks completely different. So if I say "OFFICIAL", I also have a reference to a policy in which to interpret that. That might might look like:

```
<prop name="marking-policy" ns="https://fedramp.gov/ns/oscal">Executive Order 13556</prop>
<prop name="marking" ns="https://fedramp.gov/ns/oscal">Controlled Unclassified Information</prop>
```

or 

```
<prop name="marking-policy" ns="https://fedramp.gov/ns/oscal">Australian Protective Security Policy Framework</prop>
<prop name="marking" ns="https://fedramp.gov/ns/oscal">OFFICIAL</prop>
```

**Comment by wendellpiez** [comment]
Adding these @name values for properties in the metadata (or indeed elsewhere) makes sense to me.

**Comment by david-waltermire-nist** [comment]
This is addressed in PR #619 where we added a marking property that can be used anywhere a property is allowed.
